pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -634,7 +754,7 @@ public PinotDataType getSingleValueType() {
   }
 
   public static PinotDataType getPinotDataType(FieldSpec fieldSpec) {
-    FieldSpec.DataType dataType = fieldSpec.getDataType();
+    DataType dataType = fieldSpec.getDataType();
```

Comment:
The semantics look confusing w.r.t the use of INTEGER_ARRAY v/s PRIMITIVE_INT_ARRAY

FieldSpec for a MV column is mapped to INTEGER_ARRAY and not PRIMITIVE_INT_ARRAY for PinotDataType
However, ColumnDataType is mapped to PRIMITIVE_INT_ARRAY

Reply:
We expect `Object[]` in `GenericRow` when ingesting data, but use primitive array internally for query execution. I'll rename the method and put some comments describing the behavior